### PR TITLE
Make sharrif XHTML/XHTML5 compliant

### PR DESCRIPTION
**Pull request by theseer**
This small PR adjusts the way the list elements are created to support XHTML/XHTML5.

When the browser is in XML mode, likely because the content has been served as ```application/xhtml+xml``` or due to the XML header, the strings passed to jQuery's ```append()``` and ```prepend()``` methods must be parseable as XML. 

The change by this PR is simply closing all elements directly, as all other elements are going to be added as children as needed.

Tested in Firefox 52(Linux) and Chrome 57(Linux). I did not spot any visual side effects.

I also moved the init code into a callback to be run when the document is ready rather than having it execute upon loading the script.

### CODE REVIEW

File: src/js/shariff.js [hunk 1]
```diff
@@ -257,9 +257,12 @@ module.exports = Shariff;
 // export Shariff class to global (for non-Node users)
 global.Shariff = Shariff;
 
-// initialize .shariff elements
-$('.shariff').each(function() {
-    if (!this.hasOwnProperty('shariff')) {
-        this.shariff = new Shariff(this);
-    }
+$(global.document).ready(function () {
```
Comment (liayn) [inline src/js/shariff.js]: This syntax for the ready function is deprecated in the newer versions of jquery.
One shall use: `$(function() {});` only.

Comment (theseer) [inline src/js/shariff.js]: As if jQuery code wouldn't be implicit enough yet ;)

While I will of course change it, making it a separate fix won't work as for the XHTML/XHTML5 support this change in execution is a prerequisite.

Comment (liayn) [inline src/js/shariff.js]: This is actually not about xhtml, but about race conditions. You may not access DOM until it's ready. And you never know when it is ready => Hence, your change is a dedicated bugfix.

**Comment by liayn** [comment]
@theseer can you please split the domready handler change into a dedicated PR? The remaining stuff should be easy to merge, but the domready needs adjustments in case ./dom is used instead of jQuery. @glaszig can maybe help with this?

**Comment by glaszig** [comment]
for the dom-ready function to work like with jquery #226 needs to be merged first.

**Comment by theseer** [comment]
If I see that correctly, there won't be a need for the jQuery based call anymore, so i could simply revert that change and rebase it?

**Comment by liayn** [comment]
@theseer Not sure if I can follow your reasoning. IMO this is still necessary. @glaszig only implemented a wrapper, such that we do not need to care whether jQuery is present or not at this point.

**Comment by theseer** [comment]
Okay, let me try this a different way ;) as I'm not sure what exactly I should do now.

Let's assume I split the PR - by moving the jquery dom-ready code into a separate one, then this PR if merged won't work stand alone. The second PR would have to be merged as well. But, given that the (new) wrapper should be used, I'm not sure what the second PR, implementing it the "old" way, would be good for?

Hence my suggestion to simply revert the dom-ready change I made as it won't be needed in its current form. A new Patch could then be developed to use the newer API. Once that has landed, my original PR (minus the reverted change as described above), can be merged.

**Comment by liayn** [comment]
> then this PR if merged won't work stand alone.
Why?

> But, given that the (new) wrapper should be used, ... implementing it the "old" way
Nothing changes in this regard. There is no "old" way!
What has been implemented is a tiny drop-in replacement for jQuery for sites that do not want to use jQuery, or do not need it. Nothing changes in the way we access the DOM (`$` is still there).

The dom-ready change shall be a dedicated PR, which fixes the issue that shariff may access the DOM even though it is not ready yet. The #226 PR only needs to be merged first, because otherwise we have a temporary (even more) broken code, as it adds the DOM-ready function for our drop-in replacement.

This very PR should (independently) add the XHTML compatibility.

**Comment by glaszig** [comment]
my 2 cents, @theseer ;)

the only thing you intended to change was xml closing tag compliance.

but you also wrapped the shariff element initialization at the bottom of the script in a jquery-like dom-ready handler. this is fine and will allow shariff to be included in the `<head>` section and then, when `DOMContentLoaded` triggers, initialize shariff elements. but this improvement does not really belong here.

**Comment by theseer** [comment]
I know I made two changes at once. It was with good intentions though: For whatever reason I was under the assumption that a browser in XML-Parsing mode would choke on the attempt to modify the DOM before it finished parsing the document. I probably mangled that up with some other race condition and mentally only mapped that to XML mode.

Change is coming up.

**Comment by theseer** [comment]
Looks like rebasing this to master was a bad idea.

**Comment by theseer** [comment]
Okay, I have no idea what I did wrong here to mess up the PR.
Unless anyone has a pointer what i should do to fix this one, I could also close it and open a new PR from a freshly forked master.

**Comment by glaszig** [comment]
i guess you missed syncing your master with upstream before rebasing. if it's easier for you, just do a new pr, i'd say.

**Comment by sehkunde** [comment]
Hey, before you start this anew could you please elaborate as to why self closing tags in JS strings are necessary? Self closing `<a />` or `<span />` don't seem more valid to me.

Additionally I'm not aware of a use-case for XHTML to be parsed as XML in a web browser user-agent if delivered with content-type `application/xhtml+xml` (the `<?xml>` prefix doesn't trigger the xml parser in any browser or does it?)

**Comment by liayn** [comment]
This PR looks more than ugly... we add a load of commits, but only change the DOMready thingy and the description talks something different.

Please completely abandon this PR, create a new patch based on master and fix the DOMready issue and push as new PR. Thanks a lot.

The discussion about self-closing stuff can then be done in a new issue/PR as well.

**Comment by theseer** [comment]
As already mentioned, i managed to somehow mess up the PR history when rebasing. I'll close it and open a new one for the domready and - if it's still valid - another one for the xhtml compliance.

**Pull request closed without merging**